src/Extension/Entity/ExtensionCollection.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OAT\DependencyResolver\Extension\Entity;
+
+class ExtensionCollection extends \ArrayObject implements \JsonSerializable
+{
+    /** @var Extension[] */
+    private $extensions = [];
+
+    public function add(Extension $extension): self
+    {
+        $this->offsetSet($extension->getExtensionName(), $extension);
+
+        return $this;
+    }
+
+    public function offsetSet($index, $newval)
```

Comment:
Why are you keep track of the index? If it's a numeric index, why is it important? It should be an associative array as : extensionName => extension. It doesn't have to be an ArrayObject, and you can get rid of the offset... methods. What do you think?

Reply:
Refactored with IteratorAggregate.